src/main/core/choo-choo-train.ts

```diff
@@ -161,6 +206,24 @@ export class ChooChooTrain extends Loggable {
       'We are the conductor of the choo-choo train. Running all available work in queue'
     )
 
+    const start = performance.now()
+
+    // Both server and clients will have the same data due to being provided from git,
+    // thus we obtain the data from the conductor's first job before the loop
+    const conductorWork = this.workQueue?.[0]
```

Comment:
How sure are we that this won't be a race condition? Will the first job always be finished by the time this executes?

Reply:
Whenever a ChooChooTrain instance gets created, it queues up the current package it's seeing, so every ChooChooTrain has at minimum one job upon creation. The `doWork` function is only ran by the conductor/server, so we know that it truly is the first thing to get processed. We are doing this before anything actually gets consumed from the queue, so we should be good.

https://github.com/ibm-telemetry/telemetry-js/pull/288/files#diff-08470a1af7f29045faff47e7f18077811d3c8334458d2458303d9953844480e1R69